enhancements/kubelet/user-namespaces-support.md

```diff
@@ -0,0 +1,477 @@
+
+#### Feature Gates and Sets
+
+There are three feature gates of note for this enhancement: `UserNamespacesSupport`, `UserNamespacesPodSecurityStandards`, and `ProcMountType`.
+All three of these feature gates are upstream Kubernetes features.
+
+- `UserNamespacesSupport` is the main toggle for allowing user namespaces in Kubernetes. Without it, the kubelet and kube-apiserver will filter the `hostUsers` field from a pod spec.
+- `UserNamespacesPodSecurityStandards` is for toggling the relaxation of Pod Security Standards when a pod is in a user namespace. This feature will remain in alpha, until the oldest supported kubelet
+  of the newest kube-apiserver supports denying a pod when the feature is enabled but the kernel doesn't support user namespaces.
+- `ProcMountType` is a feature for allowing an `Unmasked` `/proc` mount in a container. While not directly related to user namespaces on the surface, the feature relies on the presence of a user namespace.
+  When paired with a user namespace, the feature opens the opportunity for a container to mount `/proc` mounts internally, which is useful for running nested containers.
+
+Despite being in beta in 1.30, `UserNamespacesSupport` is not on by default, and thus won't be in 4.17. It is the main featuregate of this feature.
+Enabling `UserNamespacesSupport`, `UserNamespacesPodSecurityStandards` and `ProcMountType` will move a cluster into TechPreviewNoUpgrade in 4.17.
+
+`UserNamespacesPodSecurityStandards` is an auxilliary feature gate that loosens Pod Security Standards (PSS) for pods with user namespaces.
+At the time of writing, this relaxation is restricted to the runAsUser/runAsGroup/runAsNonRoot fields.
+However, there are [other](https://github.com/kubernetes/kubernetes/pull/126163) relaxations proposed.
+Since this feature gate is responsible for relaxing validation done on pods that have user namespaces, it's path to GA needs to be done more carefully.
+For instance, Kubernetes supports n-3 skew between the kube-apiserver and kubelet. If the kube-apiserver supports user namespaces, but the kubelet doesn't,
+then the kube-apiserver may expect a pod is confined by a user namespace when it is not, and accidentally give it more privileges than it deserves.
+Thus, `UserNamespacesPodSecurityStandards` is currently in alpha, and will not advance to GA until three releases after Kubelet supports the `UserNamespacesSupport`
+feature gate.
+This feature gate is also the main feature gate used to toggle the SCC changes above. This is done because the same version skew issue exists in Openshift,
+and it would be better to keep UserNamespacesSupport on by default (not skewing from upstrem).
+A consequence of this is the clusterpolicycontroller will also need to be taught to be aware of the `UserNamespacesPodSecutiyStandards` feature.
```

Comment:
How about adding a bit more context about the ClusterPolicyController?

```suggestion
A consequence of this is the clusterpolicycontroller will also need to be taught to be aware of the `UserNamespacesPodSecutiyStandards` feature, in order to consider these relaxations when assigning a PSa label based on the assigned SCC (if the relaxations are applicable).
```

Reply:
updated!